crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -239,49 +246,57 @@ async fn delete(
         .map_err(|_| unexpected_error!("Something went wrong"))?;
     if context_ids.is_empty() {
-        conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
-            diesel::update(dsl::default_configs)
-                .filter(dsl::key.eq(&key))
-                .set((
-                    dsl::last_modified_at.eq(Utc::now().naive_utc()),
-                    dsl::last_modified_by.eq(user.get_email()),
-                ))
-                .execute(transaction_conn)?;
+        let resp =
+            conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+                diesel::update(dsl::default_configs)
+                    .filter(dsl::key.eq(&key))
+                    .set((
+                        dsl::last_modified_at.eq(Utc::now().naive_utc()),
+                        dsl::last_modified_by.eq(user.get_email()),
+                    ))
+                    .execute(transaction_conn)?;
 
-            let deleted_row =
-                diesel::delete(dsl::default_configs.filter(dsl::key.eq(&key)))
-                    .execute(transaction_conn);
-            match deleted_row {
-                Ok(0) => Err(not_found!("default config key `{}` doesn't exists", key)),
-                Ok(_) => {
-                    let version_id = add_config_version(&state, tags, transaction_conn)?;
-                    log::info!(
-                        "default config key: {key} deleted by {}",
-                        user.get_email()
-                    );
-                    Ok(HttpResponse::NoContent()
-                        .insert_header((
-                            AppHeader::XConfigVersion.to_string(),
-                            version_id.to_string(),
-                        ))
-                        .finish())
-                }
-                Err(e) => {
-                    log::error!("default config delete query failed with error: {e}");
-                    Err(unexpected_error!("Something went wrong."))
+                let deleted_row =
+                    diesel::delete(dsl::default_configs.filter(dsl::key.eq(&key)))
+                        .execute(transaction_conn);
+                match deleted_row {
+                    Ok(0) => {
+                        Err(not_found!("default config key `{}` doesn't exists", key))
+                    }
+                    Ok(_) => {
+                        version_id = add_config_version(&state, tags, transaction_conn)?;
+                        log::info!(
+                            "default config key: {key} deleted by {}",
+                            user.get_email()
+                        );
+                        Ok(HttpResponse::NoContent()
+                            .insert_header((
+                                AppHeader::XConfigVersion.to_string(),
+                                version_id.to_string(),
+                            ))
+                            .finish())
```

Comment:
We move the response construction out of the transaction, this closure can return just the version_id.